src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -814,6 +824,23 @@ def aks_update(
     try:
         # update mc profile
         mc = aks_update_decorator.update_mc_profile_preview()
+
+        if ( raw_parameters.get("enable_azuremonitormetrics") or raw_parameters.get("disable_azuremonitormetrics")):
+            subscription_id = get_subscription_id(cmd.cli_ctx)
+            instance = client.get(resource_group_name, name)
+            remove_azuremonitormetrics = False
+            if raw_parameters.get("disable_azuremonitormetrics"):
+                remove_azuremonitormetrics = True
+
+            ensure_azure_monitor_profile_prerequisites(cmd, 
+                client,
+                subscription_id,
+                resource_group_name,
+                name,
+                instance.location,
+                raw_parameters,
+                remove_azuremonitormetrics)
```

Comment:
I would suggest putting these into `update_mc` (managed_cluster_decorator.py/AKSPreviewManagedClusterUpdateDecorator).

An example would be
```
    def update_mc(self, mc: ManagedCluster) -> ManagedCluster:
        ensure_azure_monitor_profile_prerequisites(
            self.cmd,
            self.client,
            self.context.get_subscription_id(),
            self.context.get_resource_group_name(),
            self.context.get_name(),
            self.context.get_location(),
            self.__raw_parameters,
            self.context.get_disable_azure_monitor_metrics(),
        )
        return super().update_mc()
```

Reply:
agreed, this is a neater way to do it. Updated